ui/lib/core/addon/components/key-value-header.js

```diff
@@ -1,87 +1,69 @@
+ * ```
+ * @param {string} [mode=null] - Used to set the currentPath.
+ * @param {string} [baseKey=null] - Used to generate the path backward.
+ * @param {string} [currentPath=null] - requiredParam is...
+ * @param {string} [path=null] - The fallback path.
+ * @param {string} [root=null] - Used to set the secretPath.
+ * @param {boolean} [showCurrent=true] - Boolean to show the second part of the breadcrumb, ex: the secret's name.
+ * @param {boolean} [linkToPaths=true] - If true link to the path.
+ */
 
-  baseKey: null,
-  path: null,
-  showCurrent: true,
-  linkToPaths: true,
+export default class KeyValueHeader extends Component {
+  get showCurrent() {
+    return this.args.showCurrent || true;
+  }
+  get linkToPaths() {
+    return this.args.linkToPaths || true;
+  }
 
   stripTrailingSlash(str) {
     return str[str.length - 1] === '/' ? str.slice(0, -1) : str;
-  },
+  }
 
-  currentPath: computed('mode', 'path', 'showCurrent', function () {
-    const mode = this.mode;
-    const path = this.path;
+  get currentPath() {
+    const mode = this.args.mode;
+    const path = this.args.path;
     const showCurrent = this.showCurrent;
     if (!mode || showCurrent === false) {
       return path;
     }
     return `vault.cluster.secrets.backend.${mode}`;
-  }),
-  secretPath: computed(
-    'baseKey',
-    'baseKey.{display,id}',
-    'currentPath',
-    'path',
-    'root',
-    'showCurrent',
-    function () {
-      let crumbs = [];
-      const root = this.root;
-      const baseKey = this.baseKey?.display || this.baseKey?.id;
-      const baseKeyModel = encodePath(this.baseKey?.id);
+  }
 
-      if (root) {
-        crumbs.push(root);
-      }
-
-      if (!baseKey) {
-        return crumbs;
-      }
-
-      const path = this.path;
-      const currentPath = this.currentPath;
-      const showCurrent = this.showCurrent;
-      const ancestors = utils.ancestorKeysForKey(baseKey);
-      const parts = utils.keyPartsForKey(baseKey);
-      if (ancestors.length === 0) {
-        crumbs.push({
-          label: baseKey,
-          text: this.stripTrailingSlash(baseKey),
-          path: currentPath,
-          model: baseKeyModel,
-        });
-
-        if (!showCurrent) {
-          crumbs.pop();
-        }
+  get secretPath() {
+    let crumbs = [];
+    const root = this.args.root;
+    const baseKey = this.args.baseKey?.display || this.args.baseKey?.id;
+    const baseKeyModel = encodePath(this.args.baseKey?.id);
 
-        return crumbs;
-      }
+    if (root) {
+      crumbs.push(root);
+    }
 
-      ancestors.forEach((ancestor, index) => {
-        crumbs.push({
-          label: parts[index],
-          text: this.stripTrailingSlash(parts[index]),
-          path: path,
-          model: ancestor,
-        });
-      });
+    if (!baseKey) {
+      return crumbs;
+    }
 
+    const path = this.args.path;
+    const currentPath = this.currentPath;
```

Comment:
small nit: are the variable declarations on lines 58-60 helpful here? 🤔  I might be alone thinking this - but readability might be easier knowing throughout this method where each variable is coming from further down (i.e. `baseKey` is a `const` defined within this method, but `currentPath` is actually a getter for the component) - it doesn't seem like we save a ton of cleanup by renaming them to variables. just a thought!

Reply:
Agreed. I struggle with this on glimmerizing. How much do I refactor as well as glimmer? This was a direct copy from the previous function (I didn't change any of the functionality). I realize the change here is very low risk, but I've been bit a couple times combing refactoring and glimmer together. Again, really not sure. Let me sit with this until tomorrow and I'll take another look. It's not a big deal, it's more how do I want to think about this as a rule going forward.